src/main/java/appeng/block/AEBaseBlock.java

```diff
@@ -132,27 +133,21 @@ public BlockRenderInfo getRendererInstance()
 			return this.renderInfo;
 		}
 
-		try
-		{
-			final BaseBlockRender renderer = this.getRenderer().newInstance();
-			this.renderInfo = new BlockRenderInfo( renderer );
+		final BaseBlockRender<? extends AEBaseBlock, ? extends AEBaseTile> renderer = this.getRenderer();
+		this.renderInfo = new BlockRenderInfo( renderer );
 
-			return this.renderInfo;
-		}
-		catch( InstantiationException e )
-		{
-			throw new IllegalStateException( "Failed to create a new instance of an illegal class " + this.getRenderer(), e );
-		}
-		catch( IllegalAccessException e )
-		{
-			throw new IllegalStateException( "Failed to create a new instance of " + this.getRenderer() + " because of permissions.", e );
-		}
+		return this.renderInfo;
 	}
 
+	/**
+	 * Factory method to create a new render instance.
+	 *
+	 * @return the newly created instance.
+	 */
 	@SideOnly( Side.CLIENT )
-	protected Class<? extends BaseBlockRender> getRenderer()
+	protected BaseBlockRender<? extends AEBaseBlock, ? extends AEBaseTile> getRenderer()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1452)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1452)